### ISSUE TITLE
partitioning: ATen call_function helpers ignore symbolic_trace 'tensor_meta', report 0 FLOPs/weights for nn.Linear and torch.matmul

### ISSUE THREAD

**Issue by Ravenwater** [branes-ai/graphs#63]
## Symptom

Tracing a bare `nn.Linear` (or `torch.matmul`) module through
`graphs.frontends.trace_and_partition(...)` returns a partition report
where ``total_flops == 0`` and ``total_weight_bytes == 0``. The analyzer
then predicts zero compute time and zero weight bytes for the layer --
silently.

## Repro

```python
from validation.model_v4.workloads.linear import build_linear
from graphs.frontends import trace_and_partition
from graphs.hardware.resource_model import Precision

w = build_linear(B=64, IN=1024, OUT=1024, dtype="fp32")
fx, report = trace_and_partition(w.model, w.inputs[0], precision=Precision.FP32)
sg = report.subgraphs[0]
print(sg.total_flops, sg.total_weight_bytes)  # 0  0   <-- should be ~134M, ~4.2MB
```

## Root cause

`symbolic_trace` lowers `nn.Linear.__call__` to a `call_function` node
with target `<built-in function linear>` (i.e., `torch.nn.functional.linear`).
The arg tensors carry shape via `meta['tensor_meta']` (a TensorMetadata
dataclass).

`torch.export.export` lowers it to a `call_function` node with target
`aten.linear.default`. The arg tensors carry shape via `meta['val']`
(a FakeTensor with `.numel()`).

PR #56's `_aten_weight_bytes` and `_arg_numel` helpers (and the existing
`_flops_linear_aten`, `_flops_conv2d_aten`, `_flops_addmm_aten`) all
check **only** `meta['val']`:

```python
# src/graphs/transform/partitioning/fusion_partitioner.py
@staticmethod
def _arg_numel(node, idx: int) -> int:
    if idx >= len(node.args):
        return 0
    arg = node.args[idx]
    if not hasattr(arg, 'meta') or 'val' not in arg.meta:
        return 0   # <-- silently returns 0 for symbolic_trace path
    try:
        return int(arg.meta['val'].numel())
    except Exception:
        return 0
```

The same pattern repeats in `_flops_linear_aten` (line ~700-770),
`_flops_conv2d_aten` (line ~636), `_flops_addmm_aten`. All silently
return zero on the symbolic_trace path.

## Why it didn't show up before

The standard `UnifiedAnalyzer` flow on torchvision models traces with
symbolic_trace + `prefer_modules=True`, which keeps `nn.Conv2d`,
`nn.BatchNorm2d`, `nn.ReLU` etc. as `call_module` nodes (they're FX
leaf modules). Those go through `_compute_flops_module` /
`_compute_memory` `call_module` branches and work correctly.

`nn.Linear` is **not** an FX leaf module by default, so it traces *into*
`torch.nn.functional.linear` -- a `call_function` node. Same for
`torch.matmul`. PRs #54 / #56 added precision-aware byte counts and
ATen weight extraction, but tested only with Dynamo export (which sets
`meta['val']`), not symbolic_trace (which sets `meta['tensor_meta']`).

## Affected analyzer paths

Any model with a top-level `nn.Linear` or `torch.matmul` call traced
via the default `prefer_modules=True` path. ResNet-18, MobileNet etc.
are unaffected because their Linear is preceded by Conv/Pool which
trace fine; the Linear at the head produces zero predicted FLOPs but
its contribution to total latency is small.

ViT / transformer models would be more affected (many Linears in
attention + MLP), but they also use `MultiheadAttention` which goes
through the call_module path with custom flop counting -- so the
visible drift is muted.

The harness in v4 hits this directly: every workload is exactly one
call_function Linear or matmul, so every prediction is zero.

## Fix sketch

Extract a small helper:

```python
def _meta_numel(arg) -> int:
    """Return numel from FX node arg meta, supporting both
    Dynamo (meta['val'].numel()) and symbolic_trace (meta['tensor_meta'].shape)."""
    if not hasattr(arg, 'meta'):
        return 0
    if 'val' in arg.meta:
        try:
            return int(arg.meta['val'].numel())
        except Exception:
            pass
    if 'tensor_meta' in arg.meta:
        try:
            shape = arg.meta['tensor_meta'].shape
            n = 1
            for d in shape:
                n *= int(d)
            return n
        except Exception:
            pass
    return 0
```

Use it in `_arg_numel`, `_flops_linear_aten` (replace `weight_node.meta['val'].shape` reads with `_meta_shape(weight_node)`), `_flops_conv2d_aten`, `_flops_addmm_aten`.

## Test plan

Reuse `validation/model_v4/workloads/linear.py` and `matmul.py`; assert
`total_flops > 0` and `total_weight_bytes > 0` for a small Linear
traced through `trace_and_partition`. The current test suite would have
caught this if it had a single-Linear regression test; it currently
exercises ResNet/ViT where Linear's zero contribution is masked.

## Workaround in v4 harness

The v4-3 harness will compute expected FLOPs and bytes via
`validation.model_v4.sweeps.classify.op_footprint` (closed-form
formula from the workload shape), not via the partition report. That
sidesteps this bug for the harness's own validation but does not fix
the analyzer for production use.

## Provenance

Spotted while smoke-testing `validation/model_v4/workloads/linear.py`
during V4-2 implementation (epic #62).
